tensorflow/core/grappler/optimizers/auto_mixed_precision_test.cc

```diff
@@ -0,0 +1,479 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
```

Comment:
In general, can you please use  EvaluateNodes(...) to run the original and modified graphs and verify that the outputs agree to the accuracy you expect.

Example:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/arithmetic_optimizer_test.cc#L3006
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/arithmetic_optimizer_test.cc#L3060

Reply:
Would a TODO be acceptable for this PR? These tests were written in the style of loop_optimizer_test.cc and were not designed for numerical analysis. In addition to requiring a rewrite, adding numerical checks will also likely require careful changes to the test graphs to ensure they reflect real-world numerics.
Numerical correctness is also already checked in the Python tests.

Reply:
I think simply making sure the graphs run after the rewrite would be useful. A TODO is fine for using more real-world numerics.

Reply:
Done.
I threw in numerical checks too, with a TODO noting that they should be made more realistic.

Reply:
Thanks!